### PR TITLE
Removes dead code, unused imports, adds comments and added additional info to workflow.md

### PR DESCRIPTION
**Pull request by Mapo-inf**

### CODE REVIEW

**Comment by JoelNiklaus** [comment]
Thank you.

File: scrc/preprocessors/extractors/spider_specific/section_splitting_functions.py [hunk 1]
```diff
@@ -1650,24 +1518,6 @@ def CH_BPatG(decision: Union[bs4.BeautifulSoup, str], namespace: dict) -> Option
...
-        lines = soup.split('\n\n')
-        for element in lines:
-            element = element.replace('  ', ' ')
-            paragraph = clean_text(element)
-            if paragraph not in ['', ' ', None]:
-                paragraphs.append(paragraph)
-        return paragraphs
-
-    paragraphs = get_paragraphs(decision)
+    paragraphs = get_paragraphs_unified(decision)
```
Comment (JoelNiklaus) [inline scrc/preprocessors/extractors/spider_specific/section_splitting_functions.py:1521]: did you test if this works equally well?

File: scrc/preprocessors/extractors/spider_specific/section_splitting_functions.py [hunk 2]
```diff
@@ -1467,45 +1373,7 @@ def ZH_Verwaltungsgericht(decision: Union[bs4.BeautifulSoup, str], namespace: di
...
-                        paragraph = text
-                    heading = None  # reset heading
-                # only clean and append non-empty paragraphs
-                if paragraph not in ['', ' ', None]:
-                    paragraph = clean_text(paragraph)
-                    paragraphs.append(paragraph)
-        return paragraphs
-
-    paragraphs = get_paragraphs(decision)
+    paragraphs = get_paragraphs_unified(decision)
```
Comment (JoelNiklaus) [inline scrc/preprocessors/extractors/spider_specific/section_splitting_functions.py:1376]: did you test if this works equally well?

**Review by JoelNiklaus — COMMENTED** [review]
Thank you. Could you please explain the removal of some get_paragraphs functions?

File: scrc/preprocessors/extractors/spider_specific/section_splitting_functions.py (hunk 2 above)
Comment (Mapo-inf) [reply scrc/preprocessors/extractors/spider_specific/section_splitting_functions.py:1376]: Yes I tested it. Also, the pattern extractor was always using the get_paragraphs_unified method to extract the patterns which means that it works.

Comment (JoelNiklaus) [reply scrc/preprocessors/extractors/spider_specific/section_splitting_functions.py:1376]: Great, thanks

**Pull request merged**